### DISCUSSION TITLE
Is ScriptX multithread-safe all the time?

### DISCUSSION THREAD

**Discussion by yqs112358** [Tencent/ScriptX#24] [Q&A] [answered]
Because of technical problems, in our project, **a deteached thread** is created to wait for messages posted to a **Global engine**, and this thread dispatch them to corresponding engines which handles data actually. 
So I am curious that is it safe to call functions in a **Different engine** like `engine->eval(xxx)`  or call function in `Local<Function>` generated from this **Different engine** when it is running other code at the same time , so-called "multithread-safe all the time".
Thanks very much! @LanderlYoung

**Comment by LanderlYoung** [comment, marked as answer]
A Local<> must be called in the engineScope that it blongs.

Engine scope itself holds lock, so it's safe to entet the same engine in different threadss.
